pallets/creator-staking/src/lib.rs

```diff
@@ -0,0 +1,748 @@
+
+            ensure!(amount > Zero::zero(), Error::<T>::CannotUnstakeZero);
+            ensure!(Self::is_creator_active(space_id), Error::<T>::NotRegisteredCreator,);
+
+            let current_era = Self::current_era();
+            let mut staker_info = Self::staker_info(&staker, space_id);
+            let mut stake_info =
+                Self::creator_stake_info(space_id, current_era).unwrap_or_default();
+
+            let value_to_unstake =
+                Self::unstake_from_creator(&mut staker_info, &mut stake_info, amount, current_era)?;
+
+            // Update the chunks and write them to storage
+            let mut ledger = Self::ledger(&staker);
+            ledger.unbonding_info.add(UnlockingChunk {
+                amount: value_to_unstake,
+                unlock_era: current_era + T::UnbondingPeriodInEras::get(),
+            });
+            // This should be done AFTER insertion since it's possible for chunks to merge
+            ensure!(
+                ledger.unbonding_info.len() <= T::MaxUnlockingChunks::get(),
+                Error::<T>::TooManyUnlockingChunks
+            );
+
+            Self::update_ledger(&staker, ledger);
+
+            // Update total staked value in era.
+            GeneralEraInfo::<T>::mutate(current_era, |value| {
+                if let Some(x) = value {
+                    x.staked = x.staked.saturating_sub(value_to_unstake)
+                }
+            });
+            Self::update_staker_info(&staker, space_id, staker_info);
+            CreatorEraStake::<T>::insert(space_id, current_era, stake_info);
+
+            Self::deposit_event(Event::<T>::Unstaked {
+                who: staker,
+                creator: space_id,
+                era: current_era,
+                amount: value_to_unstake,
+            });
+
+            Ok(().into())
+        }
+
+        #[pallet::call_index(5)]
+        #[pallet::weight(Weight::from_ref_time(10_000))]
+        pub fn withdraw_unstaked(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+            Self::ensure_pallet_enabled()?;
+            let staker = ensure_signed(origin)?;
+
+            let mut ledger = Self::ledger(&staker);
+            let current_era = Self::current_era();
+
+            let (valid_chunks, future_chunks) = ledger.unbonding_info.partition(current_era);
+            let withdraw_amount = valid_chunks.sum();
+
+            ensure!(!withdraw_amount.is_zero(), Error::<T>::NothingToWithdraw);
+
+            // Get the staking ledger and update it
+            ledger.locked = ledger.locked.saturating_sub(withdraw_amount);
+            ledger.unbonding_info = future_chunks;
+
+            Self::update_ledger(&staker, ledger);
+            GeneralEraInfo::<T>::mutate(current_era, |value| {
+                if let Some(x) = value {
+                    x.locked = x.locked.saturating_sub(withdraw_amount)
+                }
+            });
+
+            Self::deposit_event(Event::<T>::WithdrawnFromClaimed {
```

Comment:
Need to come up with a better error name. Because "claimed" is not relewant to ustake/withdraw flow. Claim is related to rewards.
How about `WithdrawnFromActiveCreator`